packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -1,19 +1,128 @@
 import Web3 from "web3";
 import winston from "winston";
+import { getAbi, ExpandedERC20Web3 } from "@uma/contracts-node";
+
+import { InsuredBridgeL1Client, InsuredBridgeL2Client, Deposit } from "@uma/financial-templates-lib";
+
+enum RelayerMode {
+  OnlySlow,
+  OnlySpeedUp,
+  Any,
+}
+
+enum RelayType {
+  Slow,
+  Fast,
+  No,
+}
 
 export class Relayer {
   /**
    * @notice Constructs new Relayer Instance.
    * @param {Object} logger Module used to send logs.
    * @param {Object} web3 Provider from Truffle/node to connect to Ethereum network.
+   * @param {Object} l1Client Client for fetching L1 data from the insured bridge pool and admin contracts.
+   * @param {Object} l2Client Client for fetching L2 deposit data.
+   * @param {Array} whitelistedRelayL1Tokens List of whitelisted L1 tokens that the relayer supports.
+   * @param {string} account Unlocked web3 account to send L1 messages.
+   * @param {RelayerMode} relayerMode Mode the relayer is placed in which influences how the relayer behaves.
    */
-  constructor(readonly logger: winston.Logger, readonly web3: Web3, readonly l1Client: any, readonly l2Client: any) {}
+  constructor(
+    readonly logger: winston.Logger,
+    readonly web3: Web3,
+    readonly l1Client: InsuredBridgeL1Client,
+    readonly l2Client: InsuredBridgeL2Client,
+    readonly whitelistedRelayL1Tokens: string[],
+    readonly account: string,
+    readonly relayerMode: RelayerMode = RelayerMode.Any
+  ) {}
+
+  // TODO: consider refactoring this further into multiple methods. One to build the object of relayableDeposits and one
+  // that processes relayableDeposits and calls respective relaying functions.
+  async checkForPendingDepositsAndRelay() {
+    this.logger.debug({ at: "Relayer", message: "Checking for pending deposits and relaying" });
+    const relayableDeposits: { [key: string]: any } = {};
+    for (const l1Token of this.whitelistedRelayL1Tokens) {
```

Comment:
nit: might be worth running these promises in parallel.

Reply:
ye, I think there are a few optimisations. I want to keep things as simple, short and clean as posible for now. we can always improve speed later on.

With that said, there are no promises in this loop. all methods calls in this particular loop are synchronous.